src/wp-includes/post.php

```diff
@@ -5751,39 +5751,45 @@ function get_page_by_path( $page_path, $output = OBJECT, $post_type = 'page' ) {
  * @return WP_Post|array|null WP_Post (or array) on success, or null on failure.
  */
 function get_page_by_title( $page_title, $output = OBJECT, $post_type = 'page' ) {
-	global $wpdb;
 
-	if ( is_array( $post_type ) ) {
-		$post_type           = esc_sql( $post_type );
-		$post_type_in_string = "'" . implode( "','", $post_type ) . "'";
-		$sql                 = $wpdb->prepare(
-			"
-			SELECT ID
-			FROM $wpdb->posts
-			WHERE post_title = %s
-			AND post_type IN ($post_type_in_string)
-		",
-			$page_title
-		);
-	} else {
-		$sql = $wpdb->prepare(
-			"
-			SELECT ID
-			FROM $wpdb->posts
-			WHERE post_title = %s
-			AND post_type = %s
-		",
-			$page_title,
-			$post_type
-		);
+	$last_changed = wp_cache_get_last_changed( 'posts' );
+
+	$hash      = md5( $page_title . serialize( $post_type ) );
+	$cache_key = "get_page_by_title:$hash:$last_changed";
+	$cached    = wp_cache_get( $cache_key, 'posts' );
+
+	if ( false !== $cached ) {
+		// Special case: '0' is a bad `$page_path`.
+		if ( '0' === $cached || 0 === $cached ) {
+			return null;
+		} else {
+			return get_post( $cached, $output );
+		}
 	}
 
-	$page = $wpdb->get_var( $sql );
+	// Make sure that we even query post statuses that are excluded from search
+	$post_status = array_values( get_post_stati( array( 'exclude_from_search' => true ) ) );
```

Comment:
Is this to match the query all post statuses per the old query?

Reply:
I don't know I copied this from the patch I am refreshing

Reply:
OK. I've looked at WP_Query to see what is happening.

I can also see how you are testing it in `test_should_get_different_post_statuses` and the tests pass both before and after so it appears to be safe.